crates/rpc-server/src/metrics.rs

```diff
@@ -0,0 +1,75 @@
+use gw_otel::metric::{
+    counter::Counter, encoding::text::Encode, family::Family, gauge::Gauge, registry::Registry,
+    Lazy,
+};
+
+use crate::registry::Request;
+
+pub static RPC_METRICS: Lazy<RPCMetrics> = Lazy::new(|| {
+    let metrics = RPCMetrics::default();
+    let mut registry = gw_otel::metric::global();
+    metrics.register(registry.sub_registry_with_prefix("rpc"));
+    metrics
+});
+
+#[derive(Default)]
+pub struct RPCMetrics {
+    execute_transactions: Family<ExecutionLabel, Counter>,
+    in_queue_requests: Family<RequestLabel, Gauge>,
+}
+
+#[derive(Clone, Hash, PartialEq, Eq, Encode)]
+pub enum RequestKind {
+    Tx,
+    Withdrawal,
+}
+
+impl RPCMetrics {
+    pub fn execute_transactions(&self, exit_code: i8) -> Counter {
+        self.execute_transactions
+            .get_or_create(&ExecutionLabel { exit_code })
+            .clone()
+    }
+
+    pub(crate) fn in_queue_requests(&self, req: &Request) -> Gauge {
+        let kind = match req {
+            Request::Tx(_) => RequestKind::Tx,
+            Request::Withdrawal(_) => RequestKind::Withdrawal,
+        };
+
+        self.in_queue_requests
+            .get_or_create(&RequestLabel { kind })
+            .clone()
+    }
+
+    fn register(&self, registry: &mut Registry) {
+        registry.register(
+            "execute_transactions",
+            "Number of execute_transaction requests",
+            Box::new(self.execute_transactions.clone()),
+        );
+        registry.register(
+            "in_queue_requests",
+            "Number of in queue requests",
+            Box::new(self.in_queue_requests.clone()),
+        );
```

Comment:
Read-only nodes don't have this metric.

Reply:
Yes. We won't see them in prometheus, because we don't initialize this metric for readonly node.

Reply:
Are you sure? Isn't this always registered when `RPC_METRICS` is initialized?

Reply:
They are "registered" but never create. They won't show up in prometheus side.

Reply:
What does it mean that they are never “create”?

I just tested it and if a metrics is registered, it will be scraped (by the `scrape` function).

Reply:
I mean [get_or_create](https://docs.rs/prometheus-client/0.18.1/prometheus_client/metrics/family/struct.Family.html#method.get_or_create), they are not queryable.

Reply:
OK. I see that `in_queue_requests` is now a `Family`.